## Как я использовала бутстрэп?
ДЗ

Спиридонова Алина Владиславна



In [1]:
import numpy as np

In [2]:
!pip install arch

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 981.7/981.7 kB 8.5 MB/s eta 0:00:00


In [3]:
from arch.bootstrap import IIDBootstrap

Сгенерируем выборку из нормального распределения:

In [4]:
x = np.random.normal(loc=10, scale=5, size=100)

Построим доверительный интервал для оценки среднего этой выборки:

In [5]:
boot_x = IIDBootstrap(x, seed=111111)
conf_int = boot_x.conf_int(np.mean, method='percentile', reps=10000, size=0.95)
print(f"Доверительный интервал для среднего: [{conf_int[0][0]}, {conf_int[1][0]}]")

Доверительный интервал для среднего: [8.817089241244643, 10.867515892168509]


Получили, что среднее с вероятностью 95% лежит в интервале от 8.817089241244643 до 10.867515892168509

Также построим доверительный интервал для оценки 90-процентного квантиля:

In [6]:
def quantile(x):
    return np.quantile(x, 0.9)

boot_x = IIDBootstrap(x, seed=111111)
conf_int = boot_x.conf_int(quantile, method='percentile', reps=10000, size=0.95)
print(f"Доверительный интервал для 90% квантиля: [{conf_int[0][0]}, {conf_int[1][0]}]")

Доверительный интервал для 90% квантиля: [14.974053050633653, 19.362897763674873]


Получили, что 90% квантиль с вероятностью 95% лежит в интервале от 14.974053050633653 до 19.362897763674873